[Post] The New Responsive (#5378)
GoogleChrome/web.dev: src/site/content/en/blog/new-responsive/index.md

hero, the sidebar of recently viewed items, and the product grid, are all the
exact same component, with the same markup. 

<figure>
{% Video src="video/HodOHWjMnbNw56hvNASHWSgZyAf2/D4hBchz6kaPjkgx8BCmU.mp4",
autoplay=true, muted=true, playsinline=true, loop=true, controls=true %}
<figcaption><a href="https://codepen.io/una/pen/mdOgyVL">See demo on Codepen</a> (behind a flag in Canary).</figcaption>
</figure>

There are _no_ media queries used to create this entire layout, just container
queries. This allows for each product card to shift to the proper layout to fill
its space. The grid for example, uses a minmax column layout to let the elements
flow into their space, and re-layout the grid when that space is too compressed
(which means that it hit the minimum size). 

```css
.product {
  contain: layout inline-size;
}

@container (min-width: 350px) {
  .card-container {
    padding: 0.5rem 0 0;
    display: flex;
  }
  
  .card-container button {
    /* ... */
  }
}

```

When there is at least `350px` of space in the grid, the card layout goes
horizontal by being set to `display: flex`, which has a default flex-direction
of "row". 

> With less space, the product cards stack. Each product card styles itself, which
> is something that would be impossible when only able to rely on global styles.

Reviewer: ```suggestion
With less space, the product cards stack. Each product card styles itself, 
something that would be impossible with global styles alone.

```

queries and scoped styles are so important. They give you the opportunity to
silo component styles from page layout and global styles, and user styles,
enabling more resilient responsive design. This means you can now design macro
layouts using page-based media queries, including screen-spanning nuances. At
the same time using micro layouts with container queries on the components,and
add user-preference based media queries to customize user experiences based on
their unique preferences and need.

{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/uAJDDUDLcAsLzBf0a27b.png", alt="A
circle of the new responsive", width="800", height="442" %}

_This_ is the new responsive.

It's combining macro layout with micro layout, and on top of all of that, it's
taking user customization and form factor into account.

Any of these changes alone would constitute a considerable shift in how we
design for the web. But combined, they signify a really big shift in how we even
conceptualize responsive design. It's time to think about responsive design
beyond viewport size, and start considering all of these new axes for better
component-based and customized experiences.

> > The next era of responsive design is here, and you can already start to

Reviewer: This seems a bit odd as a quote.

> > The next era of responsive design is here, and you can already start to

Author: ```suggestion
The next era of responsive design is here, and you can already start to
```
